.github/workflows/test-rock-template.yaml

```diff
@@ -0,0 +1,135 @@
+name: Build and Test Rock Template
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 0 * * *"
+
+jobs:
+  validate-rockcraft-yaml:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout target repo
+        uses: actions/checkout@v3
+        with:
+          repository: canonical/rocks-template
+          ref: main
+
+      - name: Install dependencies
+        run: |
+          sudo snap install rockcraft --classic
```

Comment:
(thinking out loud... should this actually be a matrix targeting all `latest` channgels? since the workflows may use different channels, just running this test will `latest/stable` will not cover changes being introduced in higher risks. At the same time, if `latest/stable` passes this test, but `latest/edge` doesn't, would we take any action? :thinking: - happy to hear your thoughts on this)